### ISSUE TITLE
Minmum Word Error Rate Training（MWER）

### ISSUE THREAD

**Issue by ColainCYY** [tensorflow/lingvo#118]
Hi!
Is there a guide to implement MWER Training in lingvo? I have tried to write the code from scratch, but it is a really difficult to me. I'm trying the beam-srarch based N-best MWER Training as a finetune for a trained MLE model.

**Comment by drpngx** [comment]
@rprabhavalkar can't remember where we are on the open-sourcing of MWER?

**Comment by Some-random** [comment]
I'm doing something similar and an open source implementation of MWER will be really helpful for us!

**Comment by rprabhavalkar** [comment]
Thanks for the question. We are not actively working on open-sourcing the MWER implementation at the moment. That being said, I can look into this a bit more to see if this is something that we can do relatively quickly. I'll update this thread once I have more up to date information.

**Comment by zh794390558** [comment]
Watch this.

**Comment by kaituoxu** [comment]
Hi @rprabhavalkar is there any update about MWER implementation? Thx.

**Comment by qgzang** [comment]
mark

**Comment by chenjiasheng** [comment]
@ColainCYY @drpngx @Some-random @rprabhavalkar @zh794390558 @qgzang 
I have implemented mWER loss here: [https://github.com/chenjiasheng/mwer](url).

**Comment by qgzang** [comment]
> @ColainCYY @drpngx @Some-random @rprabhavalkar @zh794390558 @qgzang
> I have implemented mWER loss here: [https://github.com/chenjiasheng/mwer](url).

thx
